### PR TITLE
Four 6732/change vuejsonpretty

### PR DESCRIPTION
**Pull request by caleeli**
## Issue & Reproduction Steps
Preview Data in Screen Builder is refreshing many times when a input changes.

## Solution
Delay and debounce the refresh of data preview

## How to Test
- Create a form with some inputs and other controls
- Go to Preview
- Focus on a input text
- Type
- Note in the Preview Data panel VueJosnPretty was changed by Monaco
- Note the Preview Data is devounced (refresh after user complete to type)

https://user-images.githubusercontent.com/8028650/191998125-ad117d9a-bb43-4801-953c-15a6a32266a3.mp4

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-6732

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -412,6 +430,13 @@ export default {
     this.loadFromLocalStorage();
   },
   methods: {
+    updateDataPreview() {
+      this.previewDataStringify = JSON.stringify(this.previewData, null, 2);
+    },
```
Comment (josechirivella) [inline src/App.vue:435]: in order to remove the created on line 409 and to make this method to look nice and have the debounce functionality. I would write it like...
```suggestion
    updateDataPreview: debounce(function () {
      this.previewDataStringify = JSON.stringify(this.previewData, null, 2);
    }, 1000),
```

File: src/App.vue [hunk 2]
```diff
@@ -315,6 +324,12 @@ export default {
     WatchersPopup,
   },
   watch: {
+    previewData: {
+      deep: true,
```
Comment (josechirivella) [inline src/App.vue:328]: do we really need to watch every single property?

File: src/App.vue (hunk 1 above)
Comment (josechirivella) [reply src/App.vue:435]: if the following suggestion is approved. You can remove the created on line 409:411

**Review by josechirivella — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 2 above)
Comment (caleeli) [reply src/App.vue:328]: We need to react to any change and refresh the preview data, but I will review a better way to do it.

File: src/App.vue (hunk 1 above)
Comment (caleeli) [reply src/App.vue]: Fixed

File: src/App.vue (hunk 2 above)
Comment (caleeli) [reply src/App.vue]: Removed the watchers, and using events to refresh the preview data

File: src/App.vue [hunk 3]
```diff
@@ -412,6 +416,22 @@ export default {
     this.loadFromLocalStorage();
   },
   methods: {
+    // eslint-disable-next-line func-names
```
Comment (josechirivella) [inline src/App.vue:419]: if you were getting an eslint-error, then leave this line as it is. But, if not, you were getting an eslint warning which you can remove this line. Same goes to line 427

**Review by josechirivella — APPROVED** [review]
Only had one comment/suggestion but everything else looks good

File: src/App.vue (hunk 3 above)
Comment (caleeli) [reply src/App.vue:419]: I agree with the previous observation, it is much better (or the right way) to "debounce" this method in its definition rather that changing it during the component creation. About the `eslint-error` maybe that could be changed in a different ticket.

**Pull request merged**